core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -156,7 +156,7 @@ TableScanContext fromSnapshotId(long id) {
         caseSensitive, colStats, projectedSchema, selectedColumns, options, id, toSnapshotId);
   }
 
-  Long toSnapshotId() {
+  public Long toSnapshotId() {
```

Comment:
Both of these methods are public so we can determine snapshotsWithin in PlanScanAction

Reply:
Maybe these should be public parts of the `TableScan` API?